PowerPointLabs/PowerPointLabs/ColorPicker/Magnifier.xaml.cs

```diff
@@ -12,59 +14,127 @@ namespace PowerPointLabs.ColorPicker
     /// </summary>
     public partial class Magnifier : Window
     {
-        private const int OVERLAY_OUTLINE_SIZE = 2;
-
-        private MagnifierOverlay magnifierOverlay;
+        private MagnificationControlHost magnificationControl;
         private System.Windows.Forms.Timer timer;
+
         private IntPtr hwndMag;
-        private List<IntPtr> magFilteredWindows;
         private bool isMagInitialized;
+        
         private float magnificationFactor;
+        private Size magnifierHalfSize;
         private Size sourceSize;
         private Size sourceHalfSize;
-        private Size actualHalfSize;
 
         public Magnifier(float magnificationFactor)
         {
             InitializeComponent();
 
-            // Calculate dimensions once
-            this.magnificationFactor = magnificationFactor;
-            sourceSize.Width = (int)(Width / magnificationFactor);
-            sourceSize.Height = (int)(Height / magnificationFactor);
-            sourceHalfSize.Width = sourceSize.Width / 2;
-            sourceHalfSize.Height = sourceSize.Height / 2;
-            actualHalfSize.Width = Width / 2;
-            actualHalfSize.Height = Height / 2;
-
             timer = new System.Windows.Forms.Timer();
             timer.Interval = 100;
             timer.Tick += new EventHandler(Timer_Tick);
 
-            magnifierOverlay = new MagnifierOverlay();
-            magnifierOverlay.Loaded += MagnifierOverlay_Loaded;
-
+            this.magnificationFactor = magnificationFactor;
             Visibility = Visibility.Visible;
         }
 
+        #region Magnification HwndHost Win32 Interop
+
+        private class MagnificationControlHost : HwndHost
+        {
+            IntPtr hwndControl;
+            IntPtr hwndHost;
+            int hostWidth, hostHeight;
+
+            internal const int HOST_ID = 0x00000002;
+            internal const string WC_MAGNIFIER = "Magnifier";
```

Comment:
Constant field should come before non-constant field and should use Pascal casing.